src/ucp/core/ucp_context.c

```diff
@@ -169,9 +169,26 @@ static ucs_config_field_t ucp_context_config_table[] = {
    "Threshold for switching from eager to rendezvous protocol",
    ucs_offsetof(ucp_context_config_t, rndv_thresh), UCS_CONFIG_TYPE_MEMUNITS},
 
+  {"RNDV_INTRA_THRESH", UCS_VALUE_AUTO_STR,
+   "Threshold for switching from eager to rendezvous protocol for intra-node\n"
+   "communication.\n"
+   " auto   - Threshold is defined by UCX_RNDV_THRESH\n"
+   " 0..inf - Explicit threshold for intra-node communication, has precedence\n"
+   "          over UCX_RNDV_THRESH config",
+   ucs_offsetof(ucp_context_config_t, rndv_intra_thresh), UCS_CONFIG_TYPE_MEMUNITS},
+
+  {"RNDV_INTER_THRESH", UCS_VALUE_AUTO_STR,
+   "Threshold for switching from eager to rendezvous protocol for inter-node\n"
+   "communication.\n"
+   " auto   - Threshold is defined by UCX_RNDV_THRESH\n"
+   " 0..inf - Explicit threshold for inter-node communication, has precedence\n"
+   "          over UCX_RNDV_THRESH config",
+   ucs_offsetof(ucp_context_config_t, rndv_inter_thresh), UCS_CONFIG_TYPE_MEMUNITS},
+
```

Comment:
The separate settings for inter/intra modes can be useful not only for RNDV_THRESH but also for many system setting like transport bandwidth and latency. And creation of two additional variables for each of such cases seems not scalable for me.

I would propose to develop the new parser for such settings to let user set them in the following way or consider different more scalable way to create separate inter/intra node setting in common:

```
UCX_RNDV_THRESH=intra:1024,inter:256K
```

Reply:
Thanks, this idea looks interesting to me!

It adds a bit of complexity, but might be useful if there is real use case for this feature in other configs.

As discussed, we can even consider using YAML or JSON format for this config. Maybe YAML would be more suitable for environment variable config, because it's more concise and does not require quotes.
So the config would be defined like:
```
UCX_RNDV_THRESH=8k     // Coarse-grained config for all
UCX_RNDV_THRESH={intra:inf,inter:256k}  // Explicit fine-grained config for both
UCX_RNDV_THRESH={inter:256k}  // implicit "auto" value for intra
```
Let's discuss this topic with others to check:
1. If this feature really useful in other configs? Would it be over-engineering?
2. Should we use 3rd party library for parsing this config. YAML seems like a good candidate